### ISSUE TITLE
Is the '_softmax_layer' function in network.py redundant

### ISSUE THREAD

**Issue by yaoqi-zd** [endernewton/tf-faster-rcnn#211]
I'm reading the source code these days and I'm just wondering is the '_softmax_layer' function in network.py redundant? since the tf.nn.softmax function will apply softmax function along the last dimension by default and I don't think the reshape operator under if condition makes any difference

![img](http://ofdz11dcz.bkt.clouddn.com/%E9%80%89%E5%8C%BA_030.png)

**Comment by endernewton** [comment]
yeah there should be some way to simply it. maybe you can do it and submit a PR. :)
